tracer/src/Datadog.Trace/LibDatadog/TraceExporter.cs

```diff
@@ -49,12 +49,19 @@ public Task<bool> SendTracesAsync(ArraySegment<byte> traces, int numberOfTraces,
                 };
 
                 var responsePtr = IntPtr.Zero;
-                using var error = NativeInterop.Exporter.Send(this, tracesSlice, (UIntPtr)numberOfTraces, ref responsePtr);
-                if (!error.IsInvalid)
+                try
                 {
-                    var ex = error.ToException();
-                    _log.Error(ex, "An error occurred while sending data to the agent. Error Code: {ErrorCode}, message: {Message}", ex.ErrorCode, ex.Message);
-                    throw ex;
+                    using var error = NativeInterop.Exporter.Send(this, tracesSlice, (UIntPtr)numberOfTraces, ref responsePtr);
+                    if (!error.IsInvalid)
```

Comment:
I still find this so confusing 😅 `error.IsInvalid` means there _wasn't_ an error?!

Reply:
negation of negation, yes there wasn't any error
else our test will blow.